### ISSUE TITLE
[Feature] Allow duplicate stack names on different servers

### ISSUE THREAD

**Issue by super-qua** [mbecker20/komodo#234]
right now, if I want to add the same stack on different servers (docker socket proxy for example), I cannot reuse the same name.

I have to assign a different name, but then the _Files on server_ option does not work properly anymore.

I have to perform a workaround, which currently is specifying the Run Directory as `../<stack_name>`

I also tried assigning a different project name, but that didn't help.

ideally I'd like to be able to create stacks with the same name on different servers, and only the stack name - server combination would be required to be unique.

thanks!

**Comment by Buco7854** [comment]
+1 for this.
Assigning a different name with the correct compose project name (on creation) works tho but kind of tedious.

**Comment by mbecker20** [comment]
For now there's a lot relying on the name being unique, from webhooks to resource sync. To be honest, I think the custom run directory and project name are not a bad solution here though. Hopefully you guys see I am more keen to prioritize changes resulting in big wins / greater abilities, over some convenience.

That is why we are thinking about wholly alternative deployment models to more conveniently support repeated deployments on many servers, I do see the value there, particularly when it comes to high availability.

**Comment by super-qua** [comment]
@mbecker20 sure, no worries. totally understandable. 

what I didn't quite get though is the workaround with the project name. does this work with existing compose files on the server? 

when I tested it, I would still get the message that there was no compose file found and the stack name was in the path of the error message, not the project name. am I overlooking something here?

**Comment by Buco7854** [comment]
There is 4 things.

The project name wich only impact compose project name when you don't set it.

The compose project name.

The working directory path

And the compose file name (or something like that)


Basically if you have multiple project with same name on different serveryiu should do this:

 Let's say you want to run `nginx`.

You should set something like this

Project name: `nginx-server1forexample`
Compose project name: `nginx`
Working directory: `/path/to/your/working/directory`
Compose file name: nothing if the file is named `compose.yml` otherwise be careful that the file name is same for example `docker-compose.yml` (I suspect that's probably what you overlooked)

Hope this help.

**Comment by vakarian92** [comment]
Alternatively, we could have the ability to deploy the same stack across multiple servers. For instance, I use two Adguard instances as DNS servers, and I could easily deploy the same stack to both of them.

**Comment by super-qua** [comment]
@Buco7854 got it, thanks. didn't have it on my radar for some reason to use absolute directory paths for the Working Directory. 

with these workarounds in place, this issue is really more of a nice to have.

**Comment by taylorabranson** [comment]
This works for me, but it does create an empty folder based on the new project name inside the stacks directory. I don't mind using the workaround, but it would be nice to not have all these empty folders.

**Comment by Buco7854** [comment]
Are you sure you are using "Files on server mode"

The only directory created is the one you provided. Basically if the dir did not exist it create the dir on the server (if you are in server mode, didn't try with UI mode)

It shouldn't even be able to run if the directory is empty since it's missing compose file. It will error telling there was no file found. At this point you can create it in the info tab yourself but this step is necessary if the file previously did not exist.

I think there is a misunderstanding somewhere. Maybe sending a pic of your config and the directory tree.

**Comment by taylorabranson** [comment]
@Buco7854 Yes, I am using "Files on Server mode". Yeah, referring to the Komodo stacks directory.
For example this is what I did when adding my adguard stacks.  I give each one a unique name as required. Then I set the server, files on server mode, the run directory, and project name.

Everything works fine for both, so not sure why that extra folder is being created. Maybe I misunderstood and did some steps wrong.

Stacks:
![stacks](https://github.com/user-attachments/assets/6d6256e4-dd44-495b-97cf-add967e25860)

Setting the config (same for both, except the server id is different of course):
![adguard server 1](https://github.com/user-attachments/assets/f7d23372-b27b-4320-8eae-3bef866d730a)

These folders are then created based on the unique stack name:

![server 1 folders](https://github.com/user-attachments/assets/38b14eae-6aeb-48f8-88e2-43759c716b9b)
![server 2 folders](https://github.com/user-attachments/assets/9e165043-3d42-401d-a8ed-6e9ee165ee9d)

**Comment by super-qua** [comment]
@taylorabranson if you specify an absolute path for the Run Directory upon stack creation instead of the relative one, I think the extra folder is not created.

**Comment by taylorabranson** [comment]
@super-qua That worked. Appreciate your help!

**Comment by lutek** [comment]
some of the workaround #321

**Comment by serfriz** [comment]
Another approach would be to not give the user the ability to name the stack within Komodo, asking for the compose project name instead, and then automatically assigning the stack name based on that and the server name. Also in the GUI give the option to list stacks by stack name or compose project name.

**Comment by Buco7854** [comment]
I mean it just restrict what the user can do. I can add the server name myself if I want to and if needed.

**Comment by serfriz** [comment]
That's fair, but a toggle to display compose project names instead of komodo project names in the stacks page may calm people's OCDs 🤣

**Comment by Podolyakofs** [comment]
so what's current workaround? - i trying to connect multiple servers with same compose project name on them (~/docker/nginx, ~/docker/beszel, etc) - and each time i need to specify full path, cause relative is pinned to "unique stack name",
`../correct_name` in `Run Directory` field doesn't work cause it resolves in `incorrect_unique_name/../correct_name` and file not found error =(

and same problem when compose folder name has capital letters (stacks names can't have them)...
